typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -113,8 +123,102 @@ export class EvmHookModule extends HyperlaneModule<
+      throw new Error(
+        'Invalid targetConfig: Updating to a custom Hook address is not supported. Please provide a valid Hook configuration.',
+      );
+    }
+
+    // save current config for comparison
+    // normalize the config to ensure it's in a consistent format for comparison
+    const currentConfig = normalizeConfig(await this.read());
+
+    // Update the config
+    this.args.config = targetConfig;
+
+    // If configs match, no updates needed
+    if (configDeepEquals(currentConfig, targetConfig)) {
+      return [];
+    }
+
+    // Else, we have to figure out what an update for this Hook entails
+    // Check if we need to deploy a new Hook
+    if (
+      // if updating from an address/custom config to a proper hook config, do a new deploy
+      typeof currentConfig === 'string' ||
+      // if updating a proper hook config whose types are different, do a new deploy
+      currentConfig.type !== targetConfig.type ||
+      // if it is not a mutable Hook, do a new deploy
+      !MUTABLE_HOOK_TYPE.includes(targetConfig.type)
+    ) {
+      const contract = await this.deploy({
+        config: targetConfig,
+      });
+
+      this.args.addresses.deployedHook = contract.address;
+      return [];
+    }
+
+    let updateTxs: AnnotatedEV5Transaction[];
+
+    // obtain the update txs for each hook type
+    switch (targetConfig.type) {
+      case HookType.INTERCHAIN_GAS_PAYMASTER:
+        updateTxs = await this.updateIgpHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      case HookType.PROTOCOL_FEE:
+        updateTxs = await this.updateProtocolFeeHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      case HookType.PAUSABLE:
+        updateTxs = await this.updatePausableHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      case HookType.ROUTING:
+      case HookType.FALLBACK_ROUTING:
+        updateTxs = await this.updateRoutingHook({
+          currentConfig,
+          targetConfig,
+        });
+        break;
+      default:
+        // MERKLE_TREE, AGGREGATION and OP_STACK hooks should already be handled before the switch
+        throw new Error(`Unsupported hook type: ${targetConfig.type}`);
+    }
+
+    // Lastly, check if the resolved owner is different from the current owner
+    const owner = await Ownable__factory.connect(
+      this.args.addresses.deployedHook,
+      this.multiProvider.getProvider(this.chain),
+    ).owner();
+
+    // Return an ownership transfer transaction if required
+    if (!eqAddress(targetConfig.owner, owner)) {
+      updateTxs.push({
+        annotation: 'Transferring ownership of ownable Hook...',
+        chainId: this.domainId,
+        to: this.args.addresses.deployedHook,
+        data: Ownable__factory.createInterface().encodeFunctionData(
+          'transferOwnership(address)',
+          [targetConfig.owner],
+        ),
+      });
+    }
```

Comment:
I've thought about doing this but didn't want to overload this PR, because we can extract this out to a wider transfer ownership helper that we can reuse across all the modules instead of just in here